### PR TITLE
Logging tweaks

### PR DESCRIPTION
**Pull request by neilblair**
Instead of one script to upload two log files to logz.io (the access log and drupal log), instead I have created a script shiplog.sh which accepts arguments for shipping a specific log like so:
`/bin/bash /app/logging/scripts/shiplog.sh [LOG_NAME] [LOG_PATH] [LOG_DATE_PATTERN] [LOG_TYPE]`

So the access log is shipped like this:
`/bin/bash /app/logging/scripts/shiplog.sh "access" "/var/log/access.log" "$(date +%d/%b/%Y:)" "nginx_access"`

The cronjon.sh script now just contains calls to the shiplog.sh to ship access and drupal logs.  For a multi-site setup, the cronjob.sh could be modified to add calls to ship multiple drupal logs for each individual site.

### CODE REVIEW

**Review by omahm — COMMENTED** [review]
Looks good, posted a comment on the curl command.

File: logging/scripts/cronjob.sh [hunk 1]
```diff
@@ -67,11 +61,43 @@ else
...
+        echo "${TODAY_DRUPAL_LOG} already exists"
+    fi
+
+    # Get latest drupal log entries and ship logz.io.
+    echo "> Retrieving latest log entries from /app/log/drupal.log and writing to ${TODAY_DRUPAL_LOG}"
+    cat /app/log/drupal.log | grep "$(date +'%a, %d/%m/%Y -')" > ./drupal-latest.log
+    diff --changed-group-format='%>' --unchanged-group-format='' $TODAY_DRUPAL_LOG drupal-latest.log > drupal-new.log
+    cat drupal-new.log >> $TODAY_DRUPAL_LOG
+    echo "> Shipping latest drupal log entries to Logz.io using cURL"
+    curl -T drupal-new.log https://listener.logz.io:8022/file_upload/${LOGZ_TOKEN}/drupal
```
[the remark this answers is not in the document]
Comment (neilblair) [reply logging/scripts/cronjob.sh]: Hi Steve, this is what I've come up with:

```
http_response=$(curl -T access-new.log -s -w "%{response_code}" https://listener.logz.io:8022/file_upload/${LOGZ_TOKEN}/nginx_access)
exit_code=$?
if [ "$exit_code" != "0" ]; then
    echo "The cURL command failed with: $exit_code"
    exit 1
elif [ "$http_response" != "200" ]; then
    echo "Log shipping failed with: $http_response"
    exit 1
elif [ "$http_response" = "200" ]; then
    echo "Log shipping succeeded with: $http_response"
fi
```

My bash scripting is not the best TBH (noob-level)!  But basically if the curl command fails, the script exits.  If the curl command succeeds, but the response returned from Logz is not 200 then script also exits.  This does mean that if the access log fails to ship, then no attempt is made to ship the drupal log later in the script ...

Note that even if the script exits with a 1, platform's cron stills shows it as a success which is annoying.  So the only way to test the script is working correctly is to ssh onto the server and manually run the script to see the output.

[the remark this answers is not in the document]
Comment (omahm) [reply logging/scripts/cronjob.sh]: Looks good..  that is annoying.
A work around to the platform issues might be to fire off an email in the elseif? 
printf "From: noreply@nidirect.gov.uk\nSubject: Logz upload failed" | /usr/sbin/sendmail neil@yadda.com

[the remark this answers is not in the document]
Comment (neilblair) [reply logging/scripts/cronjob.sh]: Couldn't get sendmail to work Steve.  I'll revisit this aspect later.  For now, I have made quite a number of changes (see updated PR comments).  Could you re-review please?

File: logging/scripts/shiplog.sh [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+LOG_NAME=$1
+LOG_PATH=$2
+LOG_DATE_PATTERN=$3
+LOG_TYPE=$4
+
+# Script will be creating logs for today and removing yesterday's logs
+TODAY_DATE=$(date +%Y-%m-%d)
+YESTERDAY_DATE=$(date --date="yesterday" +%Y-%m-%d)
+
+if [ -z "$LOG_NAME" ] || [ -z "$LOG_PATH" ] || [ -z "$LOG_DATE_PATTERN" ] || [ -z "$LOG_TYPE" ]; then
```
Comment (omahm) [inline logging/scripts/shiplog.sh:29]: This is grand and just  a side not that you can use $# to check the number of params passed

if [ $# -ne 4 ]
  then
    echo "This command requires 4 parameters, $# passed"
fi

**Review by omahm — APPROVED** [review]
Nice one, looks grand to me

File: logging/scripts/shiplog.sh (hunk 2 above)
Comment (neilblair) [reply logging/scripts/shiplog.sh:29]: Ahhhh ok - nice one 👍🏻

**Pull request merged**